pallets/settlement/src/lib.rs

```diff
@@ -973,8 +1010,8 @@ decl_module! {
         }
 
         /// Root callable extrinsic, used as an internal call to execute a scheduled settlement instruction.
-        #[weight = <T as Config>::WeightInfo::execute_scheduled_instruction(*_legs_count)]
-        fn execute_scheduled_instruction(origin, id: InstructionId, _legs_count: u32) {
+        #[weight = <T as Config>::WeightInfo::execute_scheduled_instruction(*_fungible_transfers, *_nfts_transfers)]
+        fn execute_scheduled_instruction(origin, id: InstructionId, _fungible_transfers: u32, _nfts_transfers: u32) {
```

Comment:
We can add the new parameter for 6.0, but not for 5.x.

I recommend moving the code to a `base_exec...` function that take the two counts, then add a new extrinsic that takes both counts, while leaving this one to only take a `leg_count` (just pass it as the fungible count).  This old extrinsic can be renamed `execute_scheduled_instruction_old` and the code below that schedules execution can use the new extrinsic. 

We need to make sure that there isn't any scheduled calls to the old extrinsic before removing/changing it.